packages/core/src/components/Modal/ModalHeader/ModalHeader.types.ts

```diff
@@ -26,8 +31,16 @@ interface WithDescription {
 export type ModalHeaderProps = {
   /**
    * Main heading text of the modal.
-   * When supplied, would also add an aria-labelledby attribute to the modal dialog element.
+   *
+   * - If you pass a **string**, `ModalHeader` will generate an internal ID and communicate it to the parent `Modal`
+   *   so that `aria-labelledby` is set automatically (unless `Modal` receives `aria-labelledby` prop).
+   * - If you pass a **ReactNode** (such as a custom component), **you must**:
+   *   1. Assign an **`id`** to that element (or a nested element), and
+   *   2. Pass that **same `id`** as the `aria-labelledby` prop to the `Modal`.
+   *
+   * This ensures that assistive technologies know which element is the modal's title.
+   * @see [WAI-ARIA Authoring Practices for Dialog (Modal)](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#wai-ariaroles,states,andproperties)
    */
-  title: string;
+  title: string | React.ReactNode;
```

Comment:
ElementContent? Or we don't want it because it's less descriptive?

Reply:
Also, can it be an array?

Reply:
It's a good point, let's see if a request arises, and we can add it later.

I believe most of the use cases would be wrapped under a single `div`, to flex it and position it properly, and in worst case people can just use fragment.

I generally don't like the `ElementContent` rather than just saying `ReactNode`, I think it is a bit redundant.